server/server.go

```diff
@@ -1737,15 +1737,26 @@ func (s *Server) watchServicePrimaryAddrLoop(serviceName string) {
 	} else {
 		log.Warn("service primary addr doesn't exist", zap.String("service-key", serviceKey))
 	}
+	watcher := clientv3.NewWatcher(s.client)
```

Comment:
There are many codes with the same logic, the only difference is the key. How about abstracting a function for them?

Reply:
ok, I will try it later.